cli/src/main/scala/filodb.cli/CliMain.scala

```diff
@@ -159,45 +161,45 @@ object CliMain extends ArgMain[Arguments] with FilodbClusterNode {
-          require(args.hexPk.nonEmpty, "--hexPk must be defined")
-          partKeyBrAsString(args.hexPk.get)
+          require(args.hexPk.isDefined, "--hexPk must be defined")
+          partKeyBrAsString(args.hexPk())
 
         case Some("decodeChunkInfo") =>
-          require(args.hexChunkInfo.nonEmpty, "--hexChunkInfo must be defined")
-          decodeChunkInfo(args.hexChunkInfo.get)
+          require(args.hexChunkInfo.isDefined, "--hexChunkInfo must be defined")
+          decodeChunkInfo(args.hexChunkInfo())
 
         case Some("decodeVector") =>
-          require(args.hexVector.nonEmpty && args.vectorType.nonEmpty, "--hexVector and --vectorType must be defined")
-          decodeVector(args.hexVector.get, args.vectorType.get)
+          require(args.hexVector.isDefined && args.vectorType.isDefined, "--hexVector and --vectorType must be defined")
+          decodeVector(args.hexVector(), args.vectorType())
 
         case Some("timeseriesMetadata") =>
-          require(args.host.nonEmpty && args.dataset.nonEmpty && args.matcher.nonEmpty, "--host, --dataset and --matcher must be defined")
-          val remote = Client.standaloneClient(system, args.host.get, args.port)
-          val options = QOptions(args.limit, args.sampleLimit, args.everyNSeconds.map(_.toInt),
-            timeout, args.shards.map(_.map(_.toInt)), args.spread)
-          parseTimeSeriesMetadataQuery(remote, args.matcher.get, args.dataset.get,
+          require(args.host.isDefined && args.dataset.isDefined && args.matcher.isDefined, "--host, --dataset and --matcher must be defined")
+          val remote = Client.standaloneClient(system, args.host(), args.port())
+          val options = QOptions(args.limit(), args.sampleLimit(), args.everyNSeconds.map(_.toInt).toOption,
+            timeout, args.shards.map(_.map(_.toInt)).toOption, args.spread.toOption.map(Integer.valueOf))
+          parseTimeSeriesMetadataQuery(remote, args.matcher(), args.dataset(),
             getQueryRange(args), true, options)
 
         case Some("labelValues") =>
-          require(args.host.nonEmpty && args.dataset.nonEmpty && args.labelNames.nonEmpty, "--host, --dataset and --labelName must be defined")
-          val remote = Client.standaloneClient(system, args.host.get, args.port)
-          val options = QOptions(args.limit, args.sampleLimit, args.everyNSeconds.map(_.toInt),
-            timeout, args.shards.map(_.map(_.toInt)), args.spread)
-          parseLabelValuesQuery(remote, args.labelNames, args.labelFilter, args.dataset.get,
+          require(args.host.isDefined && args.dataset.isDefined && args.labelNames.isDefined, "--host, --dataset and --labelName must be defined")
+          val remote = Client.standaloneClient(system, args.host(), args.port())
+          val options = QOptions(args.limit(), args.sampleLimit(), args.everyNSeconds.map(_.toInt).toOption,
+            timeout, args.shards.map(_.map(_.toInt)).toOption, args.spread.toOption.map(Integer.valueOf))
```

Comment:
Is `Integer.valueOf` for `args.spread` necessary?  Spread is declared with `opt[Int]`....

Reply:
`QOptions` constructor takes `Option[Integer]` (don't know the reason, seems like unfounded inconsistency).